### PR TITLE
Allow playback when server uses large buffer

### LINKED ISSUE #430: audio droppouts on local connection with big buffer
When running snapserver and snapclient on the very same machine I am getting audio dropouts when specifying a large buffer like 15000ms. In this instance I used PCM (could not reproduce with ogg/flac because I cannot build as intended, see #429 ). The machine is otherwise idle.
There is no network involved, just the linux network stack on localhost.

This should not happen.
I believe the relevant piece of the log is:

```
2018-12-18 00-22-23 [Notice] Connected to ::1
2018-12-18 00-22-23 [Info] My MAC: "60:57:18:88:d7:f9", socket: 6
2018-12-18 00-22-23 [Info] ServerSettings - buffer: 15000, latency: 0, volume: 100, muted: 0
metadata:{"STREAM":"default"}
2018-12-18 00-22-23 [Info] Codec: pcm
2018-12-18 00-22-23 [Notice] sampleformat: 48000:16:2
2018-12-18 00-22-23 [Info] frames: 1440
2018-12-18 00-22-23 [Info] Failed to get chunk
2018-12-18 00-22-23 [Info] diff to server [ms]: 0.006




2018-12-18 00-22-28 [Notice] No chunk received for 5000ms. Closing ALSA.
2018-12-18 00-22-34 [Info] frames: 1440
2018-12-18 00-22-34 [Info] age > 200: -14946
2018-12-18 00-22-34 [Info] sleep < -bufferDuration/2: -14946 < -15, sleep: -14946
2018-12-18 00-22-34 [Info] sleep < -bufferDuration/2: -14946 < -15, sleep: -14916
2018-12-18 00-22-34 [Info] sleep < -bufferDuration/2: -14916 < -15, sleep: -14886
2018-12-18 00-22-34 [Info] sleep < -bufferDuration/2: -14886 < -15, sleep: -14855
-------------------
2018-12-18 00-22-49 [Info] sleep < -bufferDuration/2: -423 < -15, sleep: -393
2018-12-18 00-22-49 [Info] sleep < -bufferDuration/2: -393 < -15, sleep: -363
2018-12-18 00-22-49 [Info] sleep < -bufferDuration/2: -363 < -15, sleep: -333
2018-12-18 00-22-49 [Info] sleep < -bufferDuration/2: -333 < -15, sleep: -303
2018-12-18 00-22-49 [Info] sleep < -bufferDuration/2: -303 < -15, sleep: -273
2018-12-18 00-22-49 [Info] sleep < -bufferDuration/2: -273 < -15, sleep: -243
2018-12-18 00-22-49 [Info] sleep < -bufferDuration/2: -243 < -15, sleep: -213
2018-12-18 00-22-49 [Info] sleep < -bufferDuration/2: -213 < -15, sleep: -183
2018-12-18 00-22-49 [Info] sleep < -bufferDuration/2: -183 < -15, sleep: -153
2018-12-18 00-22-49 [Info] sleep < -bufferDuration/2: -153 < -15, sleep: -123
2018-12-18 00-22-49 [Info] sleep < -bufferDuration/2: -123 < -15, sleep: -93
2018-12-18 00-22-49 [Info] sleep < -bufferDuration/2: -93 < -15, sleep: -63
2018-12-18 00-22-49 [Info] sleep < -bufferDuration/2: -63 < -15, sleep: -33
2018-12-18 00-22-49 [Info] sleep < -bufferDuration/2: -33 < -15, sleep: -3
2018-12-18 00-22-49 [Info] Sleep -3, age: -2, bufferDuration: 30
2018-12-18 00-22-49 [Info] Chunk: -29	-29	-29	-29	1	144
2018-12-18 00-22-49 [Info] Sleep -3, age: -4783, bufferDuration: 30
2018-12-18 00-22-49 [Info] Failed to get chunk
2018-12-18 00-22-49 [Info] Sleep -3, age: -4833, bufferDuration: 30
2018-12-18 00-22-49 [Info] Failed to get chunk
2018-12-18 00-22-49 [Info] Sleep -2, age: -4863, bufferDuration: 30
2018-12-18 00-22-49 [Info] Failed to get chunk
2018-12-18 00-22-49 [Info] Sleep -2, age: -4893, bufferDuration: 30
2018-12-18 00-22-49 [Info] Failed to get chunk
2018-12-18 00-22-49 [Info] Sleep -2, age: -4922, bufferDuration: 30
2018-12-18 00-22-49 [Info] Failed to get chunk
2018-12-18 00-22-49 [Info] Sleep -2, age: -4952, bufferDuration: 30
2018-12-18 00-22-49 [Info] Failed to get chunk
2018-12-18 00-22-49 [Info] Sleep -2, age: -4982, bufferDuration: 30
2018-12-18 00-22-49 [Info] Failed to get chunk
2018-12-18 00-22-49 [Info] Sleep -2, age: -5012, bufferDuration: 30
2018-12-18 00-22-49 [Info] Failed to get chunk
2018-12-18 00-22-49 [Info] Sleep -2, age: -5042, bufferDuration: 30
2018-12-18 00-22-49 [Info] Failed to get chunk
2018-12-18 00-22-49 [Info] Sleep -2, age: -5072, bufferDuration: 30
2018-12-18 00-22-49 [Info] Failed to get chunk
2018-12-18 00-22-49 [Info] Sleep -2, age: -5102, bufferDuration: 30
2018-12-18 00-22-49 [Info] Failed to get chunk
2018-12-18 00-22-49 [Info] Sleep -2, age: -5132, bufferDuration: 30
2018-12-18 00-22-49 [Info] Failed to get chunk
2018-12-18 00-22-49 [Info] Sleep -1, age: -5162, bufferDuration: 30
2018-12-18 00-22-49 [Info] Failed to get chunk
2018-12-18 00-22-49 [Info] Sleep -1, age: -5192, bufferDuration: 30
2018-12-18 00-22-49 [Info] Failed to get chunk
2018-12-18 00-22-49 [Info] Sleep -1, age: -5222, bufferDuration: 30
2018-12-18 00-22-49 [Info] Failed to get chunk
2018-12-18 00-22-49 [Info] Sleep -1, age: -5252, bufferDuration: 30
2018-12-18 00-22-49 [Info] Failed to get chunk
2018-12-18 00-22-49 [Info] Sleep -1, age: -5281, bufferDuration: 30
2018-12-18 00-22-49 [Info] Failed to get chunk
2018-12-18 00-22-49 [Info] Sleep -1, age: -5311, bufferDuration: 30
2018-12-18 00-22-49 [Info] Failed to get chunk
2018-12-18 00-22-49 [Info] Sleep -1, age: -5341, bufferDuration: 30
2018-12-18 00-22-49 [Info] Failed to get chunk
2018-12-18 00-22-49 [Info] Sleep -1, age: -5371, bufferDuration: 30
2018-12-18 00-22-49 [Info] Failed to get chunk
2018-12-18 00-22-49 [Info] Sleep -1, age: -5401, bufferDuration: 30
2018-12-18 00-22-49 [Info] Failed to get chunk
2018-12-18 00-22-49 [Info] Sleep -1, age: -5431, bufferDuration: 30
2018-12-18 00-22-49 [Info] Failed to get chunk
2018-12-18 00-22-49 [Info] Sleep 0, age: -5461, bufferDuration: 30
2018-12-18 00-22-49 [Info] Failed to get chunk
2018-12-18 00-22-49 [Info] Sleep 0, age: -5491, bufferDuration: 30
2018-12-18 00-22-49 [Info] Failed to get chunk
2018-12-18 00-22-49 [Info] Sleep 0, age: -5521, bufferDuration: 30
2018-12-18 00-22-49 [Info] Failed to get chunk
2018-12-18 00-22-49 [Info] Sleep 0, age: -5551, bufferDuration: 30
2018-12-18 00-22-49 [Info] Failed to get chunk
2018-12-18 00-22-49 [Info] Sleep 0, age: -5581, bufferDuration: 30
2018-12-18 00-22-49 [Info] Failed to get chunk
2018-12-18 00-22-49 [Info] Sleep 0, age: -5611, bufferDuration: 30
2018-12-18 00-22-49 [Info] Failed to get chunk
2018-12-18 00-22-49 [Info] Sleep 0, age: -5641, bufferDuration: 30
2018-12-18 00-22-49 [Info] Failed to get chunk
2018-12-18 00-22-49 [Info] Sleep 0, age: -5670, bufferDuration: 30
2018-12-18 00-22-49 [Info] Failed to get chunk
2018-12-18 00-22-49 [Info] Sleep 0, age: -5700, bufferDuration: 30
2018-12-18 00-22-49 [Info] Failed to get chunk
2018-12-18 00-22-49 [Info] Sleep 0, age: -5730, bufferDuration: 30
2018-12-18 00-22-49 [Info] Failed to get chunk
2018-12-18 00-22-49 [Info] Sleep 0
2018-12-18 00-22-49 [Info] Failed to get chunk
2018-12-18 00-22-49 [Info] age > 200: -5790
2018-12-18 00-22-49 [Info] sleep < -bufferDuration/2: -5790 < -15, sleep: -5790
2018-12-18 00-22-49 [Info] sleep < -bufferDuration/2: -5790 < -15, sleep: -5760
2018-12-18 00-22-49 [Info] sleep < -bufferDuration/2: -5760 < -15, sleep: -5730
2018-12-18 00-22-49 [Info] sleep < -bufferDuration/2: -5730 < -15, sleep: -5700
2018-12-18 00-22-49 [Info] sleep < -bufferDuration/2: -5700 < -15, sleep: -5670
2018-12-18 00-22-49 [Info] sleep < -bufferDuration/2: -5670 < -15, sleep: -5640
2018-12-18 00-22-49 [Info] sleep < -bufferDuration/2: -5640 < -15, sleep: -5610
```

### PR DESCRIPTION
**Pull request by christf**
This Patch will adust tcp timeouts to cope with bufferMs and it will adjust rules for purging data from the stream. (let the client handle purging, the stream is just for transporting data)

This fixes #430

### CODE REVIEW

File: client/stream.cpp [hunk 1]
```diff
@@ -73,7 +73,7 @@ void Stream::clearChunks()
 
 void Stream::addChunk(msg::PcmChunk* chunk)
 {
-	while (chunks_.size() * chunk->duration<cs::msec>().count() > 10000)
+	while (chunks_.size() * std::chrono::duration_cast<cs::usec>(chunk->duration<cs::msec>()) > bufferMs_ * 10 )
```
Comment (christf) [inline client/stream.cpp]: While this patch more closely resembles the current approach of the Stream class, I do not feel it is correct to discard chunks in the stream class itself.

A stream should be just that. Some other entity should handle discarding of chunks. How do you folks feel about this? I would be glad to entirely remove the check

**Comment by christf** [comment]
adjusted the code towards my comments wrt stream handling.

**Comment by christf** [comment]
removing the server patch from this patchset for now

**Comment by badaix** [comment]
Thanks a lot for your contribution, sorry again, you caught me in a moment where I could not spend time into this project.
I think most of the addressed issues have been solved, so I will close this PR.

**Pull request closed without merging**